src/controllers/trust.details.controller.ts

```diff
@@ -0,0 +1,99 @@
+import { NextFunction, Request, Response } from 'express';
+import * as config from '../config';
+import { logger } from '../utils/logger';
+import { getApplicationData, setExtraData } from '../utils/application.data';
+import * as mapperToPageService from '../utils/trust/mapper.to.page';
+import * as mapperToSessionService from '../utils/trust/mapper.to.session';
+import { ApplicationData } from '../model/application.model';
+import * as PageModel from '../model/trust.page.model';
+import { Trust, TrustKey } from '../model/trust.model';
+
+const TRUST_DETAILS_TEXTS = {
+  title: 'Tell us about the trust',
+};
+
+const get = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): void => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    const appData: ApplicationData = getApplicationData(req.session);
+
+    const trustId = req.params['id'];
+    const pageData: PageModel.TrustDetails = mapperToPageService.mapDetail(
+      appData[TrustKey]?.find(trust => trust.trust_id === trustId),
+    );
+
+    const templateName = config.TRUST_DETAILS_PAGE;
+
+    return res.render(
+      templateName,
+      {
+        backLinkUrl: '/',
+        templateName,
+        pageParams: {
+          title: TRUST_DETAILS_TEXTS.title,
+        },
+        pageData,
+      },
+    );
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+const post = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    const trustId = req.params['id'];
+
+    //  get trust data from session
+    const appData: ApplicationData = getApplicationData(req.session);
+    const trusts: Trust[] = appData[TrustKey] ?? [];
+    const trustIdx = trusts.findIndex(trust => trust.trust_id === trustId);
+
+    if (trustIdx >= 0) {
```

Comment:
could you also add a comment above line 64 similar to the other steps in this function, just to say something like 
`// remove trust from the array if it is present in the trusts array`